### PR TITLE
Optional request timeout

### PR DESCRIPTION
**Pull request by aarsilv**
🎟️  **Ticket:** [FF-1256 - Optionally pass and use, if present, the Sample Size Calculator Run ID in traffic count data tasks](https://linear.app/eppo/issue/FF-1256/optionally-pass-and-use-if-present-the-sample-size-calculator-run-id)

Currently, our client has a hard-coded default timeout of five seconds (5000 milliseconds) to request the configuration from the CDN. If it fails, it will throw a timeout error.

image
image
image

However, some clients may want to specify the timeout. This pull request adds a new optional client configuration option, `requestTimeoutMs`, to allow upstream callers of `init()` to specify a desired timeout. If it's not specified, we continue to use our default.

This pull request also adds a helpful warning message explaining the impact of a failed initialization.

Lastly, some cleanup around package version compatibility and auto-generated docs have been done as well.

For example, the below results in an error when simulating six-second latency, but the error message shows the overridden three-second timeout is at play:
```js
opts = {apiKey: '<API KEY>', requestTimeoutMs: 3000};
window.eppo.init(opts).then(function() {
  console.log('Client initialized');
});
```
image

And a timeout of ten seconds is sufficient for success even with six-second latency.
```js
opts = {apiKey: '<API KEY>', requestTimeoutMs: 10000};
window.eppo.init(opts).then(function() {
  console.log('Client initialized');
});
```
image
image

### CODE REVIEW

File: .husky/pre-commit
Comment (aarsilv) [inline .husky/pre-commit:1]: Changed to be executable so it can run as a commit hook

File: docs/js-client-sdk.eppojsclient.getassignment.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 ## EppoJSClient.getAssignment() method
 
-<b>Signature:</b>
+**Signature:**
```
Comment (aarsilv) [inline docs/js-client-sdk.eppojsclient.getassignment.md:7]: The latest auto-generated docs use `**` for bolding and `_` for italic. (which I prefer in GitHub markdown anyways 📈 )

File: docs/js-client-sdk.iclientconfig.requesttimeoutms.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<!-- Do not edit this file. It is automatically generated by API Documenter. -->
+
+[Home](./index.md) &gt; [@eppo/js-client-sdk](./js-client-sdk.md) &gt; [IClientConfig](./js-client-sdk.iclientconfig.md) &gt; [requestTimeoutMs](./js-client-sdk.iclientconfig.requesttimeoutms.md)
+
+## IClientConfig.requestTimeoutMs property
```
Comment (aarsilv) [inline docs/js-client-sdk.iclientconfig.requesttimeoutms.md:5]: Autogenerated docs for newly added client configuration

File: package.json [hunk 3]
```diff
@@ -49,7 +49,7 @@
     "prettier": "^2.7.1",
     "terser-webpack-plugin": "^5.3.3",
     "testdouble": "^3.16.6",
-    "ts-jest": "^28.0.5",
+    "ts-jest": "^29.1.1",
```
Comment (aarsilv) [inline package.json:52]: `npm` was not happy installing an older version of `ts-jest` with a newer version of `jest`. Bumping appeased it.

File: package.json
Comment (aarsilv) [inline package.json:1]: Did a `yarn install` using node 16.20, as that is the min version in our common client ([source](https://github.com/Eppo-exp/js-client-sdk-common/blob/main/package.json#L11C16-L11C21))

File: src/index.ts [hunk 4]
```diff
@@ -133,23 +138,30 @@ export class EppoJSClient extends EppoClient {
...
-  // default behavior is to use a non-expiring cache.
-  // this can be overridden after initialization.
-  EppoJSClient.instance.useNonExpiringAssignmentCache();
-
-  const configurationRequestor = new ExperimentConfigurationRequestor(localStorage, httpClient);
-  await configurationRequestor.fetchAndStoreConfigurations();
+  try {
+    const axiosInstance = axios.create({
+      baseURL: config.baseUrl || constants.BASE_URL,
+      timeout: config.requestTimeoutMs || constants.REQUEST_TIMEOUT_MILLIS,
```
Comment (sameerank) [inline src/index.ts:144]: 👍

**Review by sameerank — APPROVED** [review]
Maybe I missed it in the PR description: why was this needed for the sample size calculator? Were the counts looking wrong due to SDK timeouts?

**Pull request merged**